### PR TITLE
added vertical scaling feature gate to kubelet (related unit tests too)

### PR DESCRIPTION
**Pull request by pdgetrf**
Added vertical scaling feature gate to kubelet (related unit tests too)

Tested with unit tests and manual e2e tests on local environment.

### CODE REVIEW

File: pkg/kubelet/cm/pod_container_manager_linux.go [hunk 1]
```diff
@@ -84,11 +84,13 @@ func (m *podContainerManagerImpl) EnsureExists(pod *v1.Pod) error {
...
 
-	// parse oomKillDisable flag
-	oomKillDisable := pod.ObjectMeta.Annotations[oomKillDisableAnnotation]
-	containerConfig.ResourceParameters.OomKillDisable = false
-	if oomKillDisable == "true" {
-		containerConfig.ResourceParameters.OomKillDisable = true
+	if utilfeature.DefaultFeatureGate.Enabled(kubefeatures.VerticalScaling) {
+		// parse oomKillDisable flag
+		oomKillDisable := pod.ObjectMeta.Annotations[oomKillDisableAnnotation]
+		containerConfig.ResourceParameters.OomKillDisable = false
```
Comment (vinaykul) [inline pkg/kubelet/cm/pod_container_manager_linux.go:90]: Just checking.. the default value (feature gate disabled value) stays at false?

File: pkg/kubelet/kuberuntime/labels_test.go [hunk 2]
```diff
@@ -214,11 +217,11 @@ func TestContainerAnnotations(t *testing.T) {
 		ContainerPorts:            containerPorts,
 		PodDeletionGracePeriod:    pod.DeletionGracePeriodSeconds,
 		PodTerminationGracePeriod: pod.Spec.TerminationGracePeriodSeconds,
-		Hash:                      kubecontainer.HashContainer(container),
-		HashZeroResources:         kubecontainer.HashContainerZeroResources(container),
-		RestartCount:              restartCount,
-		TerminationMessagePath:    container.TerminationMessagePath,
-		PreStopHandler:            container.Lifecycle.PreStop,
+		Hash:                   kubecontainer.HashContainer(container),
```
Comment (vinaykul) [inline pkg/kubelet/kuberuntime/labels_test.go:220]: should the alignment stay as before (aligned to PodTerminationGracePeriod)

**Review by vinaykul — APPROVED** [review]
LTGM. a few minor comments.

File: pkg/kubelet/cm/pod_container_manager_linux.go (hunk 1 above)
Comment (pdgetrf) [reply pkg/kubelet/cm/pod_container_manager_linux.go:90]: yes

File: pkg/kubelet/kuberuntime/labels_test.go (hunk 2 above)
Comment (pdgetrf) [reply pkg/kubelet/kuberuntime/labels_test.go:220]: done by vim go plugin.

**Pull request merged**